packages/auth-construct/src/construct.ts

```diff
@@ -108,32 +121,56 @@ export class AmplifyAuth
     );
 
     // UserPool - Identity Providers
-    const providerSetupResult = this.setupIdentityProviders(
+    this.providerSetupResult = this.setupIdentityProviders(
       this.userPool,
       props.loginWith
     );
-    this.oauthMappings = providerSetupResult.oauthMappings;
 
-    // UserPool Client
+    this.domainPrefix = props.loginWith.externalProviders?.domainPrefix ?? '';
+    if (
+      this.providerSetupResult.providersList.length > 0 &&
+      this.domainPrefix
```

Comment:
should we check providers list here?
if I  explicitly provide only `domainPrefix` in inputs I'd be surprised to see it not created.
alternatively we should validate and throw if providing domain prefix doesn't make sense if providers list is empty.

Reply:
I went with the latter, throwing an error if domainPrefix is specified with an empty providers list